### PR TITLE
Full Sync :: limit to contributors and above

### PR DESCRIPTION
**Pull request by mdbitz**
Reduce included users in Full Syncs to contributor and above users. This is performed by an explicit check of the {$wpdb->prefix}user_level meta which correlates to a non 0 value.

Fixes https://wp.me/p9dueE-3JV

#### Changes proposed in this Pull Request:

* Limit the scope of User Full Syncs to contributor and above users.

#### Does this pull request change what data or activity we track or use?

NO

#### Testing instructions:

Create several users on your site, subscribers, contributors, admins, etc
Pre applying patch 
* wp shell
* $um = Automattic\Jetpack\Sync\Modules::get_module( 'full-sync' );  // get full sync module.
* $um->start( [ 'users'=>1 ] ); // initiate a full sync of users.
* $um->get_status(); // returns current status
* // Verify that the "total" for users matches # of users on the site regardless of capabilities.

Post applying patch
* wp shell
* $um = Automattic\Jetpack\Sync\Modules::get_module( 'full-sync' );  // get full sync module.
* $um->start( [ 'users'=>1 ] ); // initiate a full sync of users.
* $um->get_status(); // returns current status
* // Verify that the "total" for users matches # of users on the site with contributor or above role

### CODE REVIEW

**Comment by mdbitz** [comment]
Full Sync tests need overhaul:

1) WP_Test_Jetpack_Sync_Full_Immediately::test_full_sync_sends_all_users
  Failed asserting that 1 matches expected 11.
  
  2) WP_Test_Jetpack_Sync_Full_Immediately::test_full_sync_doesnt_send_deleted_users
  Failed asserting that 1 matches expected 2.
  
  3) WP_Test_Jetpack_Sync_Full_Immediately::test_initial_sync_doesnt_sync_subscribers
  Failed asserting that 3 matches expected 13.

File: projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-full-immediately.php [hunk 1]
```diff
@@ -1026,9 +1031,9 @@ function test_full_sync_can_sync_individual_comments() {
 	}
 
 	function test_full_sync_can_sync_individual_users() {
```
Comment (kraftbj) [inline projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-full-immediately.php:1033]: Should we add a test here to confirm that a less-than-contributor is not included?

Comment (mdbitz) [reply projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-full-immediately.php:1033]: test_full_sync_sends_all_users creates a subscriber and verifies via the count that it doesn't get included. 

I'm of the opinion that new tests have diminishing returns due to Full Sync being replaced with checksums.

**Review by leogermani — APPROVED** [review]
Tested with the provided instructions and it worked as expected.

It's worth noting that `user_level` is "semi" deprecated. Something that's (almost) not used in core anymore... but it's still there and kept up to date in the user meta table.

It's updated based on the higher level of the `level_%d` capabilities ( where `%d` goes from `0` to `10`). All [core roles still declare](https://github.com/leogermani/wordpress-develop/blob/48556.3/src/wp-admin/includes/schema.php#L747) these caps and I checked that [Woocommerce also does](https://github.com/woocommerce/woocommerce/blob/fa0464930c51a7ab34cca617f92dc27775502ee2/plugins/woocommerce/includes/class-wc-install.php#L1164).

Other custom roles might not do it and will therefore be kept out of sync.

**Comment by mdbitz** [comment]
Yes user_level is "semi" deprecated however it still provides the most reasonable level for this limitation. We can't use "edit_post" or other single capabilities as it doesn't align fully with customizations. A non 0 level still is best way to exclude subscribers and other custom roles that don't have an "edit" / contributor role.

**Pull request merged**